src/NuGet.Core/NuGet.Protocol/SourceRepository.cs

```diff
@@ -135,6 +136,7 @@ public virtual async Task<FeedType> GetFeedType(CancellationToken token)
         /// </summary>
         /// <typeparam name="T">Expected resource type</typeparam>
         /// <returns>Null if the resource does not exist</returns>
+        [Obsolete("Use the overload that takes a CancellationToken. If you don't want to support cancelation, use CancellationToken.None.")]
```

Comment:
As mentioned in the PR description, this is effectively a breaking change in our public APIs. And it's a commonly used API, so this will be "highly" impactful. However, the fix is trivial, as the message explains.

I think it's worthwhile to make this breaking change, as it's a long term win, minimizing risk of bugs similar to the two linked to this PR. Please reply if you don't think we should mark these APIs as obsolete, so as not to impact customers using the APIs in their own apps.